packages/table/src/TableHeader.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import Icon from '@leafygreen-ui/icon';
+import IconButton from '@leafygreen-ui/icon-button';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { commonCellStyles } from './styles';
+import { useTableContext, Types } from './table-context';
+import { DataType } from './utils';
+
+const thStyle = css`
+  width: 144px;
+  border-width: 0px 1px 3px 1px;
+  border-color: ${uiColors.gray.light2};
+  border-style: solid;
+`;
+
+const stickyTh = css`
+  top: 0;
+  left: 0;
+  z-index: 2;
+  position: sticky;
+`;
+
+const flexDisplay = css`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const labelStyle = css`
+  display: flex;
+  align-items: center;
+  color: ${uiColors.gray.dark2};
+  padding-right: 4px;
+`;
+
+interface TableHeaderInterface {
+  label: React.ReactElement | string;
+  onClick?: (colId: number, key: string) => void;
+  index?: number;
+  glyph?: string;
```

Comment:
These are known string literals right? Let's do that!

Reply:
good point